edge-modules/azure-monitor/tests/Microsoft.Azure.Devices.Edge.Azure.Monitor.Test/MiscellaneousTests.cs

```diff
@@ -85,6 +85,128 @@ public void TestMultipleEndpointFilter()
+
+            IEnumerable<Metric> metrics4 = PrometheusMetricsParser.ParseMetrics(testTime, prometheusMessage4, "http://VeryNoisyModule:9001/metrics");
+            IEnumerable<Metric> metrics5 = PrometheusMetricsParser.ParseMetrics(testTime, prometheusMessage5, "http://Google:9001/metrics");
+
+            bigList = metrics4.Concat(metrics5);
+
+            filter = new MetricFilter("total_network_out_bytes edgehub_gettwin_total");
+            bigList = bigList.Where(x => filter.Matches(x));
+            Assert.True(bigList.Count() == 4);
+
+            // add condition on one endpoint, but not the other
+            filter = new MetricFilter("total_network_out_bytes edgehub_gettwin_total{edge_device=\"Ubuntu-21\"}");
+            bigList = bigList.Where(x => filter.Matches(x));
+            Assert.True(bigList.Count() == 3);
+        }
+
+        // Label Equality & regex tests -- added for completeness and as a safeguard against future changes (and therefore, potential bugs)
+
+        [Fact]
+        public void TestWildcard()
+        {
+            string prometheusMessage = "edgehub_gettwin_total{iothub=\"IoTHub.azure-devices.net\",edge_device=\"Ubuntu-20\",instance_number=\"b9e51990-f3f9-4d90-8fc7-ef62d01929b2\",module_name=\"edgeHub\",ms_telemetry=\"False\"} 1683388"
+                + System.Environment.NewLine
+                + "total_network_out_bytes{iothub =\"IoTHub.azure-devices.net\",edge_device=\"Ubuntu-20\",instance_number=\"b9e51990-f3f9-4d90-8fc7-ef62d01929b2\",module_name=\"edgeAgent\",ms_telemetry=\"False\"} 5134109"
+                + System.Environment.NewLine
+                + "edgehub_gettwin_total{iothub =\"IoTHub.azure-devices.net\",edge_device=\"Ubuntu-21\",instance_number=\"023163b5-1db5-41bc-b57e-4c4d1047d9c9\",id=\"DeviceId: Ubuntu-20; ModuleId: IoTEdgeMetricsCollector [IotHubHostName: IoTHub.azure-devices.net]\"} 2"
+                + System.Environment.NewLine
+                + "max_network_out_bytes{iothub =\"IoTHub.azure-devices.net\",edge_device=\"Ubuntu-20\",instance_number=\"023163b5-1db5-41bc-b57e-4c4d1047d9c9\",id=\"DeviceId: Ubuntu-20; ModuleId: $edgeHub [IotHubHostName: IoTHub.azure-devices.net]\"} 1";
+
+            MetricFilter filter = new MetricFilter("*");
+
+            IEnumerable<Metric> metrics = PrometheusMetricsParser.ParseMetrics(testTime, prometheusMessage, "http://VeryNoisyModule:9001/metrics");
+
+            IEnumerable<Metric> result = metrics.Where(x => filter.Matches(x));
+            Assert.True(result.Count() == 4);
+
+            // Wildcards * (any characters) and ? (any single character) can be used in metric names.
+            // For example, *CPU would match maxCPU and minCPU but not CPUMaximum.
+
+            filter = new MetricFilter("*_out_bytes");
+            result = metrics.Where(x => filter.Matches(x));
+            Assert.True(result.Count() == 2);
+
+            // ???CPU would match maxCPU and minCPU but not maximumCPU.

```

Comment:
nit: modify this comment to state - "will match max_network_out_bytes, but not total_network_out_bytes"